// Copyright 2024 Sonardyne

// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the “Software”), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:

// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.

// THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.

syntax = "proto3";

import "sonardyne-public/idl/common/result.proto";
import "sonardyne-public/idl/common/uid.proto";

package sonardyne.api.pub.configuration;

message DvlUpdateRate{
  enum DvlUpdateRateEnum {
    UNKNOWN                  = 0;
    TRIGGER_RISING           = 1; // Use external trigger (rising edge)
    TRIGGER_FALLING          = 2; // Use external trigger (falling edge)
    MAX_RATE                 = 3; // Use max supported update rate
    FIXED_1HZ                = 4; // Use fixed 1Hz rate
    FIXED_2HZ                = 5; // Use fixed 2Hz rate
    FIXED_5HZ                = 6; // Use fixed 5Hz rate
    FIXED_10HZ               = 7; // Use fixed 10Hz rate
  }

  DvlUpdateRateEnum value = 1; // The type of reset to execute
  repeated DvlUpdateRateEnum valid_values = 2; // The valid values for the reset type
}

// Doppler Velocity Log configuration which controls the update rate
message DvlConfiguration {
  sonardyne.api.pub.common.Uid id = 1;                                                // Used to identify recipient instrument/module
  sonardyne.api.pub.common.Result result = 2;                                         // Populated by instrument when replying to a SetState indicating the success of the operation
  DvlUpdateRate update_rate = 3;                                                      // Specifies the update rate
}